### PR TITLE
docs: removed various slots, like avatar/icon/image from list

### PR DESCRIPTION
**Pull request by Burtan**

### CODE REVIEW

**Comment by Burtan** [comment]
See https://github.com/material-components/material-web/issues/5035

File: docs/components/list.md [hunk 1]
```diff
@@ -362,13 +362,13 @@ Token                                 | Default value
 </style>
 <md-list>
   <md-list-item headline="Wide Cat">
-    <img slot="start-image" src="https://placekitten.com/200/100">
+    <img slot="start" style="width: 40px" src="https://placekitten.com/200/100">
```
Comment (asyncLiz) [inline docs/components/list.md:365]: Can you move this into the style block instead?

```
<style>
  :root {
    ...
  }

  img {
    width: 40px;
  }
</style>
```

**Review by asyncLiz — CHANGES_REQUESTED** [review]
Thank you for this!

File: docs/components/list.md (hunk 1 above)
Comment (Burtan) [reply docs/components/list.md]: Did so.

**Review by asyncLiz — APPROVED** [review]
Looks great! Just need to sign the CLA bot: https://github.com/material-components/material-web/pull/5040#issuecomment-1741706344

**Pull request merged**